contracts/rollup/Rollup.sol

```diff
@@ -40,10 +41,7 @@ contract RollupSetup {
     uint256 public invalidBatchMarker;
 
     modifier onlyCoordinator() {
-        POB pobContract = POB(
-            nameRegistry.getContractDetails(ParamManager.proofOfBurn())
-        );
-        require(msg.sender == pobContract.getCoordinator());
+        require(msg.sender == chooser.getProposer());
```

Comment:
We now have an interface to query who's the proposer. But we need to mark the slot is proposed so that the proposer can't multi proposing in one slot.
Need to think more on this.